stl/inc/span

```diff
@@ -0,0 +1,758 @@
+
+#ifdef __cpp_lib_concepts
+// clang-format off
+template<class _It, class _Ty>
+concept _Is_span_compatible_iterator = contiguous_iterator<_It>
+                                    && is_convertible_v<remove_reference_t<iter_reference_t<_It>>(*)[], _Ty(*)[]>;
+
+template<class _It, class _Sentinel>
+concept _Is_span_compatible_sentinel = sized_sentinel_for<_Sentinel, _It>
+                                    && !is_convertible_v<_Sentinel, size_t>;
+
+template <class _Range, class _Ty>
+concept _Is_span_compatible_range = !is_array_v<remove_cvref_t<_Range>>
+                                 && !_Is_span<remove_cvref_t<_Range>>::value
+                                 && !_Is_std_array<remove_cvref_t<_Range>>::value
+                                 && ranges::contiguous_range<_Range>
+                                 && ranges::sized_range<_Range>
+                                 && (ranges::forward_range<_Range> || is_const_v<_Ty>)
+                                 && is_convertible_v<remove_reference_t<ranges::range_reference_t<_Range>>(*)[], _Ty(*)[]>;
+// clang-format on
+#else // ^^^ __cpp_lib_concepts / !__cpp_lib_concepts vvv
+
+// STRUCT TEMPLATE _Is_span_compatible_range
+// clang-format off
+template <class _Range, class _Ty>
+struct _Is_span_compatible_range : bool_constant<conjunction_v<
+    negation<is_array<_Range>>,
+    negation<_Is_span<remove_const_t<_Range>>>,
+    negation<_Is_std_array<remove_const_t<_Range>>>,
+    _Has_range_interface<_Range>,
+    _Is_span_convertible_range<_Range, _Ty>>> {};
+// clang-format on
+
+#endif // !__cpp_lib_concepts
+
+// [views.span]
+// CLASS TEMPLATE span
+template <class _Ty, size_t _Extent = dynamic_extent>
+class span : public _Span_extent_type<_Extent> {
+public:
+    using _Mybase                = _Span_extent_type<_Extent>;
+    using element_type           = _Ty;
+    using value_type             = remove_cv_t<_Ty>;
+    using size_type              = size_t;
+    using difference_type        = ptrdiff_t;
+    using pointer                = _Ty*;
+    using const_pointer          = const _Ty*;
+    using reference              = _Ty&;
+    using const_reference        = const _Ty&;
+    using iterator               = _Span_iterator<_Ty>;
+    using const_iterator         = _Span_iterator<const _Ty>;
+    using reverse_iterator       = _STD reverse_iterator<iterator>;
+    using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
+
+    static constexpr size_type extent = _Extent;
+
+    // [span.cons] Constructors, copy, and assignment
+    template <size_t _Ext = _Extent, enable_if_t<_Ext == 0 || _Ext == dynamic_extent, int> = 0>
+    constexpr span() noexcept {}
+
+#ifdef __cpp_lib_concepts
+
+    // clang-format off
+    template <class _It> requires _Is_span_compatible_iterator<_It, element_type>
+    constexpr span(_It _First, size_type _Count) noexcept // strengthened

```

Comment:
Currently these constructors are strengthened to `noexcept` despite what the standard says. Can we please continue to assume that `to_address` doesnt throw?!

Reply:
Both `to_address(T*)` and `to_address(const Ptr&)` are required to be `noexcept` by WG21-N4842 20.10.2 [memory.syn] so no assumptions are necessary. They're also implemented that way:
https://github.com/microsoft/STL/blob/8f9431931b1e1f02d24d518452ae447a843c6121/stl/inc/xutility#L135-L149

Reply:
Could I then ask someone with access to LEWG to file a bug that the 
>Throws: When and what to_­address(first) throws.

Should be removed from the specification of that constructor

Reply:
I filed an editorial issue, https://github.com/cplusplus/draft/issues/3534 . Because those paragraphs have no effect, I believe they don't require a whole LWG issue to be formally voted upon. Thanks for noticing this!